nextcord/types/protocols/base_user.py

```diff
@@ -0,0 +1,75 @@
+# The MIT License (MIT)
+# Copyright (c) 2021-present vcokltfre & tag-epic
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+from __future__ import annotations
+
+from logging import getLogger
+from typing import TYPE_CHECKING, Optional
+
+if TYPE_CHECKING:
+    from ...client.state import State
+    from ..snowflake import Snowflake
+
+
+logger = getLogger(__name__)
+
+
+class UserProtocol:
```

Comment:
This should inherit from protocol?

Reply:
Is there a Protocol class? Also, I'm not sure how to work with protocols

Reply:
Protocol classes are for structural typing

Reply:
The protocol class is typing.Protocol like the other files @Gugu7264

Reply:
Oh alright